doc/source/reference/tokenizers/token_ngram.rst

```diff
@@ -0,0 +1,237 @@
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-loose-symbol-and-loose-blank.log
+.. tokenize --tokenizer 'TokenNgram("loose_symbol", true, "loose_blank", true)' --string "0123(45)6789" --normalizer NormalizerAuto
+.. tokenize --tokenizer 'TokenNgram("loose_symbol", true, "loose_blank", true)' --string "0123-45-6789" --normalizer NormalizerAuto
+.. tokenize --tokenizer 'TokenNgram("loose_symbol", true, "loose_blank", true)' --string "0123 45 6789" --normalizer NormalizerAuto
+
+Parameters
+----------
+
+Optional parameter
+^^^^^^^^^^^^^^^^^^
+
+There are eight optional parameters.
+
+.. _token-ngram-n:
+
+``n``
+"""""
+
+This option shows ``N`` of Ngram. For example, ``n`` is ``3`` for trigram.
+
+.. _token-ngram-loose-symbol:
+
+``loose_symbol``
+""""""""""""""""
+
+Tokenize keywords including symbols, to be searched by both queries with/without symbols.
+For example, a keyword ``090-1111-2222`` will be found by any of ``09011112222``, ``090``, ``1111``, ``2222`` and ``090-1111-2222``.
+
+.. _token-ngram-loose-blank:
+
+``loose_blank``
+"""""""""""""""
+
+Tokenize keywords including blanks, to be searched by both queries with/without blanks.
+For example, a keyword ``090 1111 2222`` will be found by any of ``09011112222``, ``090``, ``1111``, ``2222`` and ``090 1111 2222``.
+
+.. _token-ngram-remove-blank:
+
+``remove_blank``
+""""""""""""""""
+
+Tokenize keywords including blanks, to be searched by queries without blanks.
+For example, a keyword ``090 1111 2222`` will be found by any of ``09011112222``, ``090``, ``1111`` or ``2222``.
+
+Note that the keyword won’t be found by a query including blanks like ``090 1111 2222``.
+
+.. _token-ngram-report-source-location:
+
+``report_source_location``
+""""""""""""""""""""""""""
+
+This option tells us a location of token of highlight target when we highlight token by ``highlight_html``.
+
+We only use this option when we want to highlight token by ``highlight_html``.
+
+When Groonga tokenize texts that target highlight, always used ``NormalizerAuto`` to normalizer until now.
+Therefore, if we use ``NormalizerNFKC100`` to normalizer, sometimes it
+can't find the position of the highlight as below.
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-no-report-source-location.log
+.. table_create Entries TABLE_NO_KEY
+.. column_create Entries body COLUMN_SCALAR ShortText
+.. table_create Terms TABLE_PAT_KEY ShortText --normalizer 'NormalizerNFKC100'
+.. column_create Terms document_index COLUMN_INDEX|WITH_POSITION Entries body
+.. load --table Entries
+.. [
+.. {"body": "ア㌕Ａz"}
+.. ]
+.. select Entries   --match_columns body   --query 'グラム'   --output_columns 'highlight_html(body, Terms)'
+
+Because we use different normalizer to normalize token.
+
+This option is its to reduce the shift of the position of the highlight as below.
+
```

Comment:
`This option is its to reduce` means `This option is used  to reduce ....'?

Reply:
Yes.
I've fixed it.